sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -778,6 +780,22 @@ private[hive] class HiveClientImpl(
   }
 
 
+
+}
+
+private[hive] object HiveClientImpl {
+  private lazy val shimDefault = IsolatedClientLoader.hiveVersion(
```

Comment:
these helper functions in Object `HiveClientImpl` are not used to interact with HiveMetaStore, they just used for run time execution , which Spark use built in `hive-exec-1.2.1.jar`.